lib/Sema/CSSimplify.cpp

```diff
@@ -4774,6 +4775,25 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
                                       flags, locatorB);
     };
 
+    // Check if any property wrappers on the base of the member lookup have
+    // mactching members that we can fall back to.
+    if (auto dotExpr =
+            dyn_cast_or_null<UnresolvedDotExpr>(locator->getAnchor())) {
+      auto baseExpr = dotExpr->getBase();
+      auto resolvedOverload = findSelectedOverloadFor(baseExpr);
+      if (auto wrappedProperty =
+              getPropertyWrapperInformation(resolvedOverload)) {
+        auto wrapperTy = wrappedProperty->second;
```

Comment:
Does this handles cases where there's a generic parameter substitution applied at the site of the member reference?

For example:

```swift
@propertyWrapper
struct S<Value, U> {
  var wrappedValue: Value
  init(initialValue: @autoclosure () -> Value) {
    self.wrappedValue = initialValue()
  }
}

extension S where U == String {
  func baz(_ u: U) {}
}

struct X<T> {
  @S<Int, T> var foo = 0
}

let x = X<String>()
x.foo.baz("")
```

I _believe_ you'll have to substitute in the type variables used to open any generic parameters within the property wrapper type.

As an alternative approach, when creating the disjunction for the choices, would it be possible to look for var overloads with attached property wrappers and add a fixed overload to the disjunction to also try the backing var? I _believe_ that would solve this problem and also handle other contextual failures for free (though I don't know if that would make the system significantly slower to solve than this approach).

Reply:
Unfortunately it doesn’t yet, we want to get basic stuff in first and iterate on that since getting contextual types in would be tricky...